mono/utils/strenc.c

```diff
@@ -41,7 +41,7 @@ static const char trailingBytesForUTF8[256] = {
  * of bytes in the returned string, not including the terminator.
  */
 gunichar2 *
-mono_unicode_from_external (const gchar *in, gsize *bytes)
+mono_unicode_from_external (const gchar *in, glong *bytes)
```

Comment:
Should we instead standardize on `gsize` and update method body to use that instead of `glong`? The problem is `glong` on 64-bit will be 8 bytes everywhere except Windows which seems odd.

Reply:
I had forgotten about that unfortunate difference. The `g_utf8_to_utf16 ` in glib accepts a `glong`, which is what cause this in the first place (https://github.com/mono/mono/blob/9be68f8952ea0e1aad582bfe2f47bad71aee2cc7/mono/eglib/giconv.c#L974).

I feel a little wary about changing that function to accept `gsize` though. What do you think?

Reply:
I would say a) ask upstream b) if we stick with `glong` is there any way to prevent size mismatch we just encountered?